#!/usr/bin/env python
# -*- coding: utf-8 -*-
PACKAGE = "radar_ros"
 
# 导入dynamic_reconfigure功能包提供的参数生成器（parameter generator）
from dynamic_reconfigure.parameter_generator_catkin import *
 
# 创建一个参数生成器
gen = ParameterGenerator()
 
# 定义动态配置的参数
 
gen.add("x_front", double_t, 0, "x_front_param", 0.0,  -20, 20)
gen.add("y_front", double_t, 0, "y_front_param", 0.0,  -20, 20)
gen.add("z_front", double_t, 0, "z_front_param", 0.0,  -20, 20)
gen.add("roll_front", double_t, 0, "roll_front_param", 0.0,  -3.14, 3.14)
gen.add("pitch_front", double_t, 0, "pitch_front_param", 0.0,  -3.14, 3.14)
gen.add("yaw_front", double_t, 0, "yaw_front_param", 0.0,  -3.14, 3.14)

gen.add("x_back", double_t, 0, "x_back_param", 0.0,  -20, 20)
gen.add("y_back", double_t, 0, "y_back_param", 0.0,  -20, 20)
gen.add("z_back", double_t, 0, "z_back_param", 0.0,  -20, 20)
gen.add("roll_back", double_t, 0, "roll_back_param", 0.0,  -3.14, 3.14)
gen.add("pitch_back", double_t, 0, "pitch_back_param", 0.0,  -3.14, 3.14)
gen.add("yaw_back", double_t, 0, "yaw_back_param", 0.0,  -3.14, 3.14)

gen.add("x_left", double_t, 0, "x_left_param", 0.0,  -20, 20)
gen.add("y_left", double_t, 0, "y_left_param", 0.0,  -20, 20)
gen.add("z_left", double_t, 0, "z_left_param", 0.0,  -20, 20)
gen.add("roll_left", double_t, 0, "roll_left_param", 0.0,  -3.14, 3.14)
gen.add("pitch_left", double_t, 0, "pitch_left_param", 0.0,  -3.14, 3.14)
gen.add("yaw_left", double_t, 0, "yaw_left_param", 0.0,  -3.14, 3.14)

gen.add("x_right", double_t, 0, "x_right_param", 0.0,  -20, 20)
gen.add("y_right", double_t, 0, "y_right_param", 0.0,  -20, 20)
gen.add("z_right", double_t, 0, "z_right_param", 0.0,  -20, 20)
gen.add("roll_right", double_t, 0, "roll_right_param", 0.0,  -3.14, 3.14)
gen.add("pitch_right", double_t, 0, "pitch_right_param", 0.0,  -3.14, 3.14)
gen.add("yaw_right", double_t, 0, "yaw_right_param", 0.0,  -3.14, 3.14)
 
# 退出
exit(gen.generate(PACKAGE, "dynamic_calib_param", "radar_ros"))

